src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -955,6 +957,51 @@ void WCSimPrimaryGeneratorAction::GeneratePrimaries(G4Event* anEvent)
       	*/
       }
 
+    } else if (useIBDEvt) {
+        // Generate IBD events from an input spectrum
+        WCSimIBDGen *IBDGen = new WCSimIBDGen(myDetector);
```

Comment:
There's a memory leak here, with this created every event and never `delete`d

I think doing something similar to the light injector code (where you have a `WCSimIBDGen *` in the class private data members, and check whether it's a `nullptr` or not to do stuff with)

Reply:
Thanks for noticing this! Sorted to follow the same logic that `LIGen` uses.